composeApp/src/commonMain/kotlin/sp/bvantur/tasky/event/data/EventRepositoryImpl.kt

```diff
@@ -36,25 +39,49 @@ class EventRepositoryImpl(
 
             is TaskyResult.Success -> {
                 if (!response.data.userExists) {
-                    TaskyResult.Error(CommunicationError.UnknownError()) // TODO better error handling
+                    TaskyResult.Error(TaskyError.UnknownError()) // TODO better error handling
                 } else {
                     localDataSource.saveAttendee(response.data.attendee.asAttendeeEntity())
                     TaskyResult.Success(
                         Attendee(
                             userId = response.data.attendee.userId,
-                            name = response.data.attendee.name
+                            name = response.data.attendee.name,
+                            email = response.data.attendee.email
                         )
                     )
                 }
             }
         }
     }
 
-    override suspend fun createEvent(event: Event): TaskyEmptyResult<CommunicationError> =
-        remoteDataSource.createEvent(event.asCreateEventRequest()).onSuccess { data ->
-            localDataSource.saveEvent(data.asEventEntity())
-            data.attendees.forEach { attendee ->
-                localDataSource.saveAttendee(attendee.asAttendeeEntity())
-            }
+    override suspend fun createEvent(event: Event): TaskyEmptyResult<TaskyError> {
+        // TODO if hostId doesn't exist we need to logout before executing this action
+        val eventEntity = event.asEventEntity(localDataSource.getHostId() ?: "")
+        val attendeeEntities = event.attendees.map {
+            it.asAttendeeEntity(
+                eventId = eventEntity.id,
+                remindAt = eventEntity.remindAt
+            )
+        }
+        val saveEventResponse = localDataSource.saveEvent(eventEntity)
+        val saveAttendeesResponse = localDataSource.saveAttendees(attendeeEntities)
```

Comment:
When you have operations closely related to each other like this, wrapping it in a transaction block would prevent an invalid state where one operation failed while the rest succeeded.